File: csharp/Original/PublishingHouse/Books/Entities/Format.cs
using PublishingHouse.Core.Validation;
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Books.Entities;

public record Format
{
    public Format(FormatType formatType, PositiveInt totalCopies, PositiveInt soldCopies)
    {
        FormatType = formatType;
        TotalCopies = totalCopies;
        SoldCopies = soldCopies;

        TotalCopies.AssertGreaterOrEqualThan(SoldCopies);
    }

    public FormatType FormatType { get; }
    public PositiveInt TotalCopies { get; private set; }
    public PositiveInt SoldCopies { get; private set; }

    public void SetTotalCopies(PositiveInt totalCopies) =>
        TotalCopies = totalCopies;

    public void SetSoldCopies(PositiveInt soldCopies) =>
        TotalCopies = soldCopies;

}

public record FormatType(string Value): NonEmptyString(Value);

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/entities/Format.java
package io.eventdriven.slimdownaggregates.original.domain.books.entities;

import io.eventdriven.slimdownaggregates.original.infrastructure.valueobjects.PositiveInt;

public class Format {
  private final FormatType formatType;
  private PositiveInt totalCopies;
  private PositiveInt soldCopies;

  public Format(FormatType formatType, PositiveInt totalCopies, PositiveInt soldCopies)
  {
    this.formatType = formatType;
    this.totalCopies = totalCopies;
    this.soldCopies = soldCopies;
  }

  public FormatType formatType() {
    return formatType;
  }

  public PositiveInt totalCopies() {
    return totalCopies;
  }

  public PositiveInt soldCopies() {
    return soldCopies;
  }

  public void setTotalCopies(PositiveInt totalCopies) {
    this.totalCopies = totalCopies;
  }

  public void setSoldCopies(PositiveInt soldCopies) {
    this.soldCopies = soldCopies;
  }
}
